### ISSUE TITLE
Did anyone try replacing max pooling with average pooling?

### ISSUE THREAD

**Issue by remindchobits** [yanx27/Pointnet_Pointnet2_pytorch#104]
After replace the torch.max with torch.mean, the evaluation accuracy is about 10% higher than the training accuracy, which is weird, did anybody ever do this and share some experiences?[
![屏幕快照 2021-04-27 15 48 47](https://user-images.githubusercontent.com/18598039/116253074-7077b080-a770-11eb-9040-50e338713386.png)

**Comment by yanx27** [comment]
Hi @remindchobits , even if you use `MaxPool`, the evaluation results will be still higher than that of training during the initial stage.

**Comment by remindchobits** [comment]
> Hi @remindchobits , even if you use `MaxPool`, the evaluation results will be still higher than that of training during the initial stage.

Thank you for the reply!

Actually the average pooling network now is on the 129th epoch, the corresponding accuracies are:

Epoch 81 (129/150):
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 308/308 [13:43<00:00,  2.67s/it]
Train Instance Accuracy: 0.712558
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 78/78 [03:26<00:00,  2.65s/it]
Test Instance Accuracy: 0.814503, Class Accuracy: 0.748164
Best Instance Accuracy: 0.820513, Class Accuracy: 0.748164

the test accuracy is still 10% higher than the training acc.

Comparing with max pooling:

2020-12-31 09:31:33,052 - Model - INFO - Epoch 129 (129/150):
2020-12-31 09:46:39,646 - Model - INFO - Train Instance Accuracy: 0.900256
2020-12-31 09:50:21,434 - Model - INFO - Test Instance Accuracy: 0.889824, Class Accuracy: 0.855844
2020-12-31 09:50:21,434 - Model - INFO - Best Instance Accuracy: 0.902644, Class Accuracy: 0.865699
2020-12-31 09:50:21,434 - Model - INFO - Epoch 130 (130/150):
2020-12-31 10:04:57,467 - Model - INFO - Train Instance Accuracy: 0.900733
2020-12-31 10:08:34,422 - Model - INFO - Test Instance Accuracy: 0.897837, Class Accuracy: 0.875054
2020-12-31 10:08:34,422 - Model - INFO - Best Instance Accuracy: 0.902644, Class Accuracy: 0.875054

which is much normal..

Checking the structure, the only speculation I can make is that for the average pooling, the same drop out rate is too high (p=0.4)

I just wanna ask whether there is other guys who tried this..
